src/cmd/src/bin/greptime.rs

```diff
@@ -62,10 +63,36 @@ enum SubCommand {
 #[global_allocator]
 static ALLOC: tikv_jemallocator::Jemalloc = tikv_jemallocator::Jemalloc;
 
-#[tokio::main]
-async fn main() -> Result<()> {
-    setup_human_panic();
-    start(Command::parse()).await
+fn main() -> Result<()> {
+    // Set the stack size to 8MB for the thread so it wouldn't crash on large stack usage in debug mode
+    // see https://github.com/GreptimeTeam/greptimedb/pull/4317
+    // and https://github.com/rust-lang/rust/issues/34283
+    let builder = std::thread::Builder::new().name("main_spawn".to_string());
+    #[cfg(debug_assertions)]
+    let builder = builder.stack_size(8 * 1024 * 1024);
+    builder
+        .spawn(|| {
+            let body = async {
+                setup_human_panic();
+                start(Command::parse()).await
+            };
+            #[allow(clippy::expect_used, clippy::diverging_sub_expression)]
+            {
+                let mut builder = tokio::runtime::Builder::new_multi_thread();
+
+                #[cfg(debug_assertions)]
+                let builder = builder.thread_stack_size(8 * 1024 * 1024);
```

Comment:
If we set the stack size of the runtime, do we need to set the stack size for the main thread?

Reply:
> If we set the stack size of the runtime, do we need to set the stack size for the main thread?

still need to. overflow happen in main thread before this fix, meaning tokio is running a large stack using main thread, builder for runtime only set stack size for newly spawned thread not the thread runtime is `block_on` anyway, so the main problem is setting larger stack size for main thread(but rust's main thread seems to be default to be 2MB), setting stack size for runtime is just a precaution

Reply:
> > If we set the stack size of the runtime, do we need to set the stack size for the main thread?
> 
> still need to. overflow happen in main thread before this fix, meaning tokio is running a large stack using main thread, builder for runtime only set stack size for newly spawned thread not the thread runtime is `block_on` anyway, so the main problem is setting larger stack size for main thread(but rust's main thread seems to be default to be 2MB), setting stack size for runtime is just a precaution(previous it's 2MB by default though, maybe setting to 4 MB is more reasonable?)